server: replace LatencyGetter with nodedialer.Dialer.Latency

asubiotto opened the pull request:
LatencyGetter was a struct used to get latencies to other nodes so that EXPLAIN
ANALYZE could display these latencies in a diagram. Unfortunately, the usage
of this resulted in a synchronous RPC during query setup, causing unnecessary
tail latencies.

Since this information is available through an RPCContext, this commit removes
the LatencyGetter and replaces its usage with a call to nodedialer.Dialer's new
Latency method, which looks up an exponentially weighted moving average latency
to a given node ID in its RPCContext.

Release note: None (no major releases with performance regression)

Closes #59093

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59131)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
LGTM but Nathan should probably take a look to make sure we're not misusing the RPCContext somehow.

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59131#-:-MRQSNiCA6lUE5EoZTKP:bxbcuw6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks!

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59131#-:-MRQeS6CEl0tkT4QMN8Y:brfqlux)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Commit:
server: replace LatencyGetter with nodedialer.Dialer.Latency

LatencyGetter was a struct used to get latencies to other nodes so that EXPLAIN
ANALYZE could display these latencies in a diagram. Unfortunately, the usage
of this resulted in a synchronous RPC during query setup, causing unnecessary
tail latencies.

Since this information is available through an RPCContext, this commit removes
the LatencyGetter and replaces its usage with a call to nodedialer.Dialer's new
Latency method, which looks up an exponentially weighted moving average latency
to a given node ID in its RPCContext.

Release note: None (no major releases with performance regression)

asubiotto commented:
Failure tracked in #59186 

TFTRs

bors r=nvanbenschoten,yuzefovich
